airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/state/WorkflowState.java

```diff
@@ -104,6 +104,27 @@ public void setCancelledForReset(final boolean cancelledForReset) {
     this.cancelledForReset = cancelledForReset;
   }
 
+  public void hasMultipleCancels() {
+    final ChangedStateEvent event = new ChangedStateEvent(
+        StateField.MULTIPLE_CANCEL,
```

Comment:
this feels like a hack to me. it seems like we need a way of handling events in such a way that it doesn't require this custom handling for every event type. let's discuss it when we sync up at 11am.